File: app/src/main/java/com/jetsynthesys/rightlife/RetrofitData/ApiClient.java
package com.jetsynthesys.rightlife.RetrofitData;



import android.content.Context;

import com.jetsynthesys.rightlife.BuildConfig;
import com.jetsynthesys.rightlife.ui.utility.LoggingInterceptor;

import java.util.concurrent.TimeUnit;

import okhttp3.OkHttpClient;
import okhttp3.logging.HttpLoggingInterceptor;
import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;




public class ApiClient {
    private static final String BASE_URL = BuildConfig.BASE_URL; // Your API URL //"https://qa.rightlife.com/api/app/api/"; // Your API URL
    private static final String BASE_URL2 = "http://18.159.113.191:8081/app/api/"; // Your API URL
    private static final String BASE_URL3 = "http://18.159.113.191:8080/app/api/"; // Your API URL

    public static final String CDN_URL_QA = BuildConfig.CDN_URL;//"https://d1sacaybzizpm5.cloudfront.net/"; // Your API URL
    public static final String CDN_URL_PROD = "https://d1uxs9zd0apq85.cloudfront.net/"; // Your API URL

    private static Retrofit retrofit = null;
    HttpLoggingInterceptor loggingInterceptor = new HttpLoggingInterceptor();

    public static Retrofit getClient(Context context) {
        HttpLoggingInterceptor loggingInterceptor = new HttpLoggingInterceptor();
        // Set logging level (BODY for detailed logging, BASIC for minimal)
        loggingInterceptor.setLevel(HttpLoggingInterceptor.Level.BODY);


        // Add the logging interceptor to OkHttpClient
        OkHttpClient client = new OkHttpClient.Builder()
                .addInterceptor(new LoggingInterceptor())
                .addInterceptor(new NetworkConnectionInterceptor(context))
                .connectTimeout(1, TimeUnit.MINUTES)
                .readTimeout(30, TimeUnit.SECONDS)
                .writeTimeout(15, TimeUnit.SECONDS)
                .retryOnConnectionFailure(true)
                .build();
        //if (retrofit == null) {
            retrofit = new Retrofit.Builder()
                    .baseUrl(BASE_URL)
                    .client(client)
                    .addConverterFactory(GsonConverterFactory.create()) // This converts JSON to Java objects
                    .build();
        //}
        return retrofit;
    }

    /*public static Retrofit getClient() {
        HttpLoggingInterceptor loggingInterceptor = new HttpLoggingInterceptor();
        // Set logging level (BODY for detailed logging, BASIC for minimal)
        loggingInterceptor.setLevel(HttpLoggingInterceptor.Level.BODY);


        // Add the logging interceptor to OkHttpClient
        OkHttpClient client = new OkHttpClient.Builder()
                .addInterceptor(new LoggingInterceptor())
                .connectTimeout(1, TimeUnit.MINUTES)
                .readTimeout(30, TimeUnit.SECONDS)
                .writeTimeout(15, TimeUnit.SECONDS)
                .retryOnConnectionFailure(true)
                .build();
        //if (retrofit == null) {
        retrofit = new Retrofit.Builder()
                .baseUrl(BASE_URL2)
                .client(client)
                .addConverterFactory(GsonConverterFactory.create()) // This converts JSON to Java objects
                .build();
        //}
        return retrofit;
    }*/

    public static Retrofit getDevClient() {
        HttpLoggingInterceptor loggingInterceptor = new HttpLoggingInterceptor();
        // Set logging level (BODY for detailed logging, BASIC for minimal)
        loggingInterceptor.setLevel(HttpLoggingInterceptor.Level.BODY);


        // Add the logging interceptor to OkHttpClient
        OkHttpClient client = new OkHttpClient.Builder()
                .addInterceptor(new LoggingInterceptor())
                .build();
            retrofit = new Retrofit.Builder()
                    .baseUrl(BASE_URL2)
                    .client(client)
                    .addConverterFactory(GsonConverterFactory.create()) // This converts JSON to Java objects
                    .build();
        return retrofit;
    }

    public static Retrofit getClient2() {
      /*  if (retrofit == null) {
            retrofit = new Retrofit.Builder()
                    .baseUrl(BASE_URL2)
                    .addConverterFactory(GsonConverterFactory.create()) // This converts JSON to Java objects
                    .build();
        }*/

        retrofit = new Retrofit.Builder()
                .baseUrl("https://api.example.com/")
                .addConverterFactory(GsonConverterFactory.create())
                .build();
        return retrofit;


    }




 /*   baseUrl: 'https://stage.rightlife.com/api',
    bobaseUrl: 'https://bo-stage.rightlife.com/api',
    cdnUrl: 'https://d2wde8gz1ymp04.cloudfront.net/',

    env: 'prod',
    baseUrl: 'https://app.rightlife.com/api',
    bobaseUrl: 'https://bo.rightlife.com/api',
    cdnUrl: 'https://d1uxs9zd0apq85.cloudfront.net/',

    env: 'qa',
    baseUrl: 'https://qa.rightlife.com/api/app/api/',
    bobaseUrl: 'https://bo-qa.rightlife.com/api',
    cdnUrl: 'https://d1sacaybzizpm5.cloudfront.net/',

    env: 'dev',
    baseUrl: 'https://api-dev.rightlife.com',
    bobaseUrl: 'https://bo-api-dev.rightlife.com/bo/api',
    cdnUrl: 'https://d3p9402jq4v8pt.cloudfront.net/',*/

}

File: app/src/main/java/com/jetsynthesys/rightlife/ai_package/data/repository/ApiClient.kt
package com.jetsynthesys.rightlife.ai_package.data.repository

import com.jakewharton.retrofit2.adapter.kotlin.coroutines.CoroutineCallAdapterFactory
import com.jetsynthesys.rightlife.BuildConfig
import kotlinx.serialization.json.Json
import okhttp3.OkHttpClient
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory
import java.util.concurrent.TimeUnit

object RetrofitClient {
    private const val BASE_URL = BuildConfig.BASE_URL
    private const val BASE_URL_FAST_API = BuildConfig.BASE_URL_AI
    private const val BASE_URL_FOOD_CAPTURE_API = "https://api.spoonacular.com/"
    private const val BASE_URL_FOOD_CAPTURE_NEW_API = "https://us-central1-snapcalorieb2bapi.cloudfunctions.net/"

    private val okHttpClient = OkHttpClient.Builder()
        .connectTimeout(40, TimeUnit.SECONDS) // Set connection timeout (default: 10s)
        .readTimeout(40, TimeUnit.SECONDS)    // Set read timeout
        .writeTimeout(40, TimeUnit.SECONDS)   // Set write timeout
        .build()

    val retrofit: Retrofit by lazy {
        Retrofit.Builder()
            .baseUrl(BASE_URL)
            .addConverterFactory(GsonConverterFactory.create())
            .build()
    }
    private val json = Json {
        ignoreUnknownKeys = true // Ignore unknown keys in JSON
        coerceInputValues = true // Coerce invalid values to default (e.g., null for missing fields)
    }
    val retrofitFastApi: Retrofit by lazy {
        Retrofit.Builder()
            .baseUrl(BASE_URL_FAST_API)
            .addConverterFactory(GsonConverterFactory.create())
            .client(okHttpClient) // Attach custom OkHttpClient with timeouts
            .build()
    }


    val retrofitFoodCaptureApi: Retrofit by lazy {
        Retrofit.Builder()
            .baseUrl(BASE_URL_FOOD_CAPTURE_API)
            .addConverterFactory(GsonConverterFactory.create())
            .build()
    }

    val retrofitFoodCaptureImageApi: Retrofit by lazy {
        Retrofit.Builder()
            .baseUrl(BASE_URL_FOOD_CAPTURE_NEW_API)
            .addConverterFactory(GsonConverterFactory.create())
            .client(okHttpClient) // Attach custom OkHttpClient with timeouts
            .build()
    }
}

object ApiClient {
    val apiService: ApiService by lazy {
        RetrofitClient.retrofit.create(ApiService::class.java)
    }

    val apiServiceFastApi: ApiService by lazy {
        RetrofitClient.retrofitFastApi.create(ApiService::class.java)
    }

    val apiServiceFoodCaptureApi: ApiService by lazy {
        RetrofitClient.retrofitFoodCaptureApi.create(ApiService::class.java)
    }

    val apiServiceFoodCaptureImageApi: ApiService by lazy {
        RetrofitClient.retrofitFoodCaptureImageApi.create(ApiService::class.java)
    }
}
